### PR TITLE
generate .env via CLI

### PR DESCRIPTION
**Pull request by georgedias**

### CODE REVIEW

**Comment by aaronlippold** [comment]
Any README updates on this so folks know the command is there.

File: docs/emasser/index.md [hunk 1]
```diff
@@ -104,6 +104,38 @@ Fo instruction on how to request an eMASS visit [eMASS Account Process Request a
 * [/api/systems/{systemId}/poams/{poamId}/milestones](#delete-milestones)
 * [/api/systems/{systemId}/artifacts](#delete-artifacts)
 
+## Generating an eMASS Configuration File
+Provided with the eMASS API CLI is an interactive command line user interface for generating the configuration file `.env` required to connect to an eMASS instance.
+
+```
+Generate a configuration file (.env) for accessing an eMASS instances.
```
Comment (em-c-rod) [inline docs/emasser/index.md:111]: `an eMASS instance` or `eMASS instances`

File: docs/emasser/index.md [hunk 2]
```diff
@@ -104,6 +104,38 @@ Fo instruction on how to request an eMASS visit [eMASS Account Process Request a
...
+
+USAGE
+  $ saf emasser configure
+
+DESCRIPTION
+  Generate a configuration file (.env) for accessing an eMASS instances.
```
Comment (em-c-rod) [inline docs/emasser/index.md:117]: Same

File: src/commands/emasser/configure.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+   ${colors.blue('EMASSER_KEY_FILE_PATH') + colors.green('    <The eMASS key.pem private encrypting the key in PEM format (file, include the path)>')}
+   ${colors.blue('EMASSER_CERT_FILE_PATH') + colors.green('   <The eMASS cert.pem containing the certificate information in PEM format (file, include the path)')}
+   ${colors.blue('EMASSER_KEY_PASSWORD') + colors.green('     <The password for the private encryption key.pem file')}
+   ${colors.yellow('The following variables are optional, if not provided defaults are used:')}
+   ${colors.blue('EMASSER_PORT') + colors.green('                <The server communication port number (default is 443)')}
+   ${colors.blue('EMASSER_REQUEST_CERT') + colors.green('        <Server requests a certificate from clients - true or false (default true)')}
+   ${colors.blue('EMASSER_REJECT_UNAUTHORIZED') + colors.green(' <Reject connection not authorized with the list of supplied CAs- true or false (default true)')}
+   ${colors.blue('EMASSER_DEBUGGING') + colors.green('           <Set debugging on or off - true or false (default false)')}
```
Comment (em-c-rod) [inline src/commands/emasser/configure.ts:21]: Set debugging on (true) or off (false) (default false)
or
Turn debugging on - true or false (default false)

File: src/commands/emasser/configure.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+import {Command} from '@oclif/core'
+import {generateConfig} from '../../utils/emasser/generateConfig'
+
+export default class EmasserBuildConfig extends Command {
+  static summary = 'Generate a configuration file (.env) for accessing an eMASS instances.'
+
+  static description =
+  `
+   ${colors.yellow('The following variables are required:')}
+   ${colors.blue('EMASSER_API_KEY_API_KEY') + colors.green('  <The eMASS API key (api-key) - valid key is > 30 alpha numeric characters>')}
```
Comment (em-c-rod) [inline src/commands/emasser/configure.ts:11]: EMASSER_API_KEY

File: src/commands/emasser/configure.ts [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+   ${colors.blue('EMASSER_API_KEY_USER_UID') + colors.green(' <The eMASS User Unique Identifier (user-uid)>')}
+   ${colors.blue('EMASSER_HOST') + colors.green('             <The Full Qualified Domain Name (FQDN) for the eMASS server>')}
```
Comment (em-c-rod) [inline src/commands/emasser/configure.ts:13]: This is fine to me

File: src/utils/emasser/generateConfig.ts [hunk 6]
```diff
@@ -0,0 +1,274 @@
+import path from 'path'
+import colors from 'colors' // eslint-disable-line no-restricted-imports
+import fse from 'fs-extra'
+import dotenv from 'dotenv'
+import inquirer from 'inquirer'
+import inquirerFileTreeSelection from 'inquirer-file-tree-selection-prompt'
+
+const PROMPT_MESSAGE = [
+  'Provide the eMASS API key (api-key) - valid key is > 30 alpha numeric characters:',
```
Comment (em-c-rod) [inline src/utils/emasser/generateConfig.ts:9]: Provide the eMASS API key (EMASS_API_KEY) - valid key is > 30 alpha numeric characters:

File: src/utils/emasser/generateConfig.ts [hunk 7]
```diff
@@ -0,0 +1,274 @@
...
+  'Provide the eMASS User unique identifier (user-uid):',
```
Comment (em-c-rod) [inline src/utils/emasser/generateConfig.ts:10]: (EMASSER_USER_UID)

File: src/commands/emasser/configure.ts (hunk 5 above)
Comment (em-c-rod) [reply src/commands/emasser/configure.ts:13]: or you can switch to EMASSER_HOST_URL

File: src/commands/emasser/configure.ts [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+   ${colors.blue('EMASSER_KEY_FILE_PATH') + colors.green('    <The eMASS key.pem private encrypting the key in PEM format (file, include the path)>')}
+   ${colors.blue('EMASSER_CERT_FILE_PATH') + colors.green('   <The eMASS cert.pem containing the certificate information in PEM format (file, include the path)')}
+   ${colors.blue('EMASSER_KEY_PASSWORD') + colors.green('     <The password for the private encryption key.pem file')}
```
Comment (em-c-rod) [inline src/commands/emasser/configure.ts:16]: EMASSER_KEY_FILE_PASSWORD

File: src/utils/emasser/generateConfig.ts [hunk 9]
```diff
@@ -0,0 +1,274 @@
...
+  'Provide the eMASS key.pem private encrypting the key in PEM format (file, include the path):',
+  'Provide the eMASS cert.pem containing the certificate information in PEM format (file, include the path):',
+  'Provide the password for the private encryption key.pem file (no validation is performed):',
+  'Provide the server communication port number (default is 443):',
+  'Server requests a certificate from clients - true or false (default true):',
+  'Reject connection not authorized with the list of supplied CAs- true or false (default true):',
+  'Set debugging on or off - true or false (default false):',
+  'Display null value fields - true or false (default true):',
+  'Convert epoch to data/time value - true or false (default false):',
+]
```
Comment (em-c-rod) [inline src/utils/emasser/generateConfig.ts:21]: Add extra prompt at the end that tells users to do `emasser get test_connection` and `emasser get -h` for more options.

File: docs/emasser/index.md (hunk 1 above)
Comment (georgedias) [reply docs/emasser/index.md:111]: done

File: src/commands/emasser/configure.ts (hunk 3 above)
Comment (georgedias) [reply src/commands/emasser/configure.ts:21]: done

**Pull request merged**